registry/regclient/profile.go

```diff
@@ -135,3 +170,48 @@ func (c Client) doJSONProfileReq(method string, p *registry.Profile) (*registry.
 	env.Data.Peername = env.Data.Username
 	return env.Data, nil
 }
+
+// RegistryResponse is a generic container for registry requests
+// TODO(dustmop): Only used currently for ProveKey
+type RegistryResponse struct {
```

Comment:
I'm also doing this for the HTTPClient work, would be good to have a common place for it with the shifting emphasis to HTTP APIs. `apiutil` definitely feels like the best place for it.

If you feel like it, you can leave a `// TODO(arqu): update with common API response object` as I'll soon be doing a large swoop of my own TODOs and can cover this too.

Reply:
I've added a PR to `apiutil` for this. Also one more note the "status" response is not strictly correct per our API specs.